server/src/main/java/org/apache/druid/rpc/indexing/OverlordClient.java

```diff
@@ -72,21 +73,28 @@ public interface OverlordClient
   ListenableFuture<Void> runTask(String taskId, Object taskObject);
 
   /**
-   * Run a "kill" task for a particular datasource and interval. Shortcut to {@link #runTask(String, Object)}.
-   *
-   * The kill task deletes all unused segment records from deep storage and the metadata store. The task runs
-   * asynchronously after the API call returns. The resolved future is the ID of the task, which can be used to
-   * monitor its progress through the {@link #taskStatus(String)} API.
-   *
-   * @param idPrefix   Descriptive prefix to include at the start of task IDs
-   * @param dataSource Datasource to kill
-   * @param interval   Interval to kill
-   *
-   * @return future with task ID
+   * Shortcut to {@link #runKillTask(String, String, Interval, Integer, DateTime)}.
+   * Note that this method is deprecated and newer implementations must use {@link #runKillTask(String, String, Interval, Integer, DateTime)}.
    */
+  @Deprecated
```

Comment:
Why deprecate it? This is not a public API, you can just get rid of the older methods. (unless there are rolling upgrade concerns, which doesn't seem to be the case since an Overlord on an old version would just get a null value of `maxUsedLastUpdated`)

Reply:
Ah, I was just following the existing pattern to maintain backwards compatibility to support rolling upgrade/downgrade scenarios and interface-extension points. I suppose it's ok to remove in this case?

Reply:
Doesn't seem like this would have any rolling upgrade concerns as you have just added a new nullable parameter that would be sent over the wire. A service on an older version would just read that param as null.

Reply:
Got it, makes sense and thanks for checking. Removed the methods